examples/stable_diffusion_v2/ldm/modules/attention.py

```diff
@@ -16,16 +16,45 @@
 
 import numpy as np
 from ldm.util import is_old_ms_version
+from packaging import version
 
 import mindspore as ms
 from mindspore import nn, ops
 from mindspore.common.initializer import initializer
 
+
+def new_version():
+    return version.parse(ms.__version__) >= version.parse("2.2")
+
+
 try:
-    from mindspore.nn.layer.flash_attention import FlashAttention
+    if new_version():
```

Comment:
A new MS version a year from now will be different from the new version today. It is better to hard-code the exact MS version inside the `if` branch as follows: 
```python
from mindone.utils.version_control import MSVersion

if MSVersion >= "2.2.0":
    ...
```

Reply:
I think inside the `new_version` function that I wrote, the exact version (2.2) has been hard-coded. It is equivalent to your solution, right?

Reply:
Yes, but not exactly. Say, for instance, if a new version 2.3 is released and it has some compatibility issues, then we would need to call the function `new_new_version` or `even_newer_version`. 😁 Hard-coding eliminates this problem and increases readability, as one can directly read the version limitation in the statement.

Reply:
I got your point and it makes sense to me. Updated.

Reply:
After I updated with [3e21d93](https://github.com/mindspore-lab/mindone/pull/291/commits/3e21d93810bd49e95e89419b8b2ae2bb5cc2740a), there is an error for ms2.1 on 910A:
```
RuntimeError: Unsupported func type for Custom operator on CPU, it should be 'hybrid', 'akg', 'pyfunc' or 'aot' or 'julia', but got [tbe] for Custom operator [Custom_flash_attention_b34108071e6ecfbd49a35ff8bb58abbeab1dfa62618a51f566f6c01f2260b47f]
```

I don't know the reason. If I undo [3e21d93](https://github.com/mindspore-lab/mindone/pull/291/commits/3e21d93810bd49e95e89419b8b2ae2bb5cc2740a), only keep [4a69ba3](https://github.com/mindspore-lab/mindone/pull/291/commits/4a69ba36b409ce1bc832b2223b145fd1efd88517), the following command is working fine:
`python text_to_image.py --prompt "a wolf in winter forest" -v "2.0"`

Reply:
I figured out that the aforementioned error was caused by calling `MSVersion >= "2.2.0"` for multiple times. I will only keep [4a69ba3](https://github.com/mindspore-lab/mindone/pull/291/commits/4a69ba36b409ce1bc832b2223b145fd1efd88517) to avoid this error on 910A.

Reply:
Interestingly, `new_version` is called multiple times as well. I will check and fix `MSVersion` later,  then. Thanks!

Reply:
I found out the reason because if I edited it like this:
```
NEW_VERSION = MSVersion >= "2.2.0"

if NEW_VERSION:
...
```
Then the inference script went well without errors. I found it weird too.